internal/types/etcd_storage.go

```diff
@@ -23,6 +25,17 @@ func ParseETCDStorageType(s string) ETCDStorageType {
 		return storageType
 	}
 
-	// TODO: we have to decide what to do in this situation
-	return defaultETCDStorageType
+	panic(fmt.Errorf("unsupported storage type %s", s))
+}
+
+// ParseETCDEndpoint returns the default ETCD endpoints used to interact with the Tenant Control Plane backing storage.
+func ParseETCDEndpoint(conf *viper.Viper) string {
+	switch ParseETCDStorageType(conf.GetString("etcd-storage-type")) {
+	case ETCD:
+		return conf.GetString("etcd-endpoints")
+	case KineMySQL:
+		return "127.0.0.1:2379"
```

Comment:
You could add the `http`protocol over here

Reply:
In this way, you could avoid the compatibility issue mentioned [here](https://github.com/clastix/kamaji/pull/66#discussion_r912681689)

Reply:
As I mentioned in the following [comment](https://github.com/clastix/kamaji/pull/66#discussion_r913448716), there's no need for protocol: I think it's safe to remove `https` from your changes unless there's a specific reason for it.